src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy

```diff
@@ -28,148 +28,141 @@ import java.nio.file.attribute.PosixFilePermissions
      * Also produces a file named [zip.gz].md5
      */
-    public static class NativeLinuxZipCompressor extends Compressor {
+    static class NativeLinuxZipCompressor extends Compressor {
 
         @Override
-        void compressFile(File from, File to, File workingDirectory = null) {
+        void compressFile(File from, File to) {
             try {
-                compressDirectory(from, to, workingDirectory);
+                compressDirectory(from, to)
             } catch (Exception ex) {
                 throw new IOException("Could not zip file ${from} to zip archive ${to}", ex)
             }
         }
 
         @Override
-        void compressDirectory(File from, File to, File workingDirectory = null) {
+        void compressDirectory(File from, File to) {
 
             try {
-                String result = LocalExecutionHelper.executeSingleCommand(getCompressionString(from, to, workingDirectory).toString());
+                String result = LocalExecutionHelper.executeSingleCommand(getCompressionString(from, to).toString())
             } catch (Exception ex) {
                 throw new IOException("Could not zip folder ${from} to zip archive ${to}", ex)
             }
         }
 
         @Override
-        void decompress(File from, File to, File workingDirectory = null) {
-            String result = LocalExecutionHelper.executeSingleCommand(getDecompressionString(from, to, workingDirectory));
-//            println(result);
-        }
-
-        private String getWorkingDirectory(File from, File workingDirectory = null) {
-            if (!workingDirectory) {
-                workingDirectory = from.getParentFile().getAbsoluteFile();
+        void decompress(File from, File to) {
+            try {
+                String result = LocalExecutionHelper.executeSingleCommand(getDecompressionString(from, to))
+            } catch (Exception ex) {
+                throw new IOException("Could not unzip zip archive ${from} to ${to}", ex)
             }
-            String wdPath = workingDirectory ? "cd ${workingDirectory.getAbsolutePath()} &&" : "";
-            return wdPath;
         }
 
         @Override
-        GString getCompressionString(File from, File to, File workingDirectory = null) {
-            String wdPath = getWorkingDirectory(from, workingDirectory);
-
+        GString getCompressionString(File from, File to) {
             String _to = to.getAbsolutePath()
-            GString zipTarget = "${_to} ${from.getName()} -x '*.svn*'"
+            GString zipTarget = "${_to} ${from.getName()}"
 
-//            GString gString = "${wdPath} [[ -e ${_to} ]] && zip -dr9 ${zipTarget} || zip -r9 ${zipTarget}";
-            GString gString = "[[ -f \"${_to}\" ]] && rm ${_to}; ${wdPath} zip -r9 ${zipTarget} > /dev/null && md5sum ${_to}";
-            return gString;
+            GString gString = "[[ -f \"${_to}\" ]] && rm ${_to}; cd ${from.parent} && zip -r9 ${zipTarget} > /dev/null && md5sum ${_to}"
```

Comment:
Then it should also be `rm "{_to}"`. With rm it would actually be particularly important, rigth?

```groovy
def _to = "/ upsSomeSpace"
```

Reply:
Generally, we should think about using single quotes filenames. Even better, use the (upcoming) Bash-escaping function/service.